godot-core/src/engine/file.rs

```diff
@@ -0,0 +1,766 @@
+    pub fn write_string(&mut self, value: impl Into<GodotString>) -> IoResult<()> {
+        self.fa.store_string(value.into());
+        self.check_error()?;
+        Ok(())
+    }
+
+    /// Writes string to the file as Pascal String.
+    ///
+    /// This function is meant to be used in binary files. To store a string in a text file,
+    /// use `store_string()`
+    ///
+    /// Pascal String is useful for writing and retrieving verying-length string data from binary files.
+    /// It is saved with length-prefix, instead of null terminator as in C strings.
+    ///
+    /// See also:
+    /// - [Wikipedia article](https://en.wikipedia.org/wiki/String_(computer_science)#Length-prefixed)
+    /// - [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html#class-fileaccess-method-store-pascal-string)
+    pub fn write_pascal_string(&mut self, value: impl Into<GodotString>) -> IoResult<()> {
+        self.fa.store_pascal_string(value.into());
+        self.check_error()?;
+        Ok(())
+    }
+
+    /// Write string to the file as a line.
+    ///
+    /// For more information about underlying method see
+    /// [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html#class-fileaccess-method-store-line).
+    pub fn write_string_line(&mut self, value: impl Into<GodotString>) -> IoResult<()> {
+        self.fa.store_line(value.into());
+        self.check_error()?;
+        Ok(())
+    }
+
+    /// Write [PackedStringArray] to the file as delimited line.
+    ///
+    /// For writing traditional `CSV` format, provide comma (`','`) as `delim`.
+    ///
+    /// For more information about underlying method see
+    /// [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html#class-fileaccess-method-store-csv-line).
+    pub fn write_csv_line(
+        &mut self,
+        values: PackedStringArray,
+        delim: impl Into<GodotString>,
+    ) -> IoResult<()> {
+        self.fa.store_csv_line_ex(values).delim(delim.into()).done();
+        self.check_error()?;
+        Ok(())
+    }
+
+    /// Write [Variant] to the file.
+    ///
+    /// If `full_objects` is set to `true`, encoding objects is allowed (and can potentially include GDScript code).
+    /// Not all properties of the Variant are included. Only properties that are exported (#[export]) will be serialized.
+    ///
+    /// For more information about underlying method see
+    /// [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html#class-fileaccess-method-store-var).
+    pub fn write_variant(&mut self, value: Variant, full_objects: bool) -> IoResult<()> {
+        self.fa
+            .store_var_ex(value)
+            .full_objects(full_objects)
+            .done();
+        self.check_error()?;
+        Ok(())
+    }
+
+    /// Set big endian-endiadiness.
```

Comment:
Typo, also comments that don't add anything over method name are generally not needed.

```suggestion
    /// Set true to use big-endian, false to use little-endian.
```

Do you also know if this can be changed in the middle of a file? This might be an interesting piece of info to add.